### DISCUSSION TITLE
Show rules on figure caption

### DISCUSSION THREAD

**Discussion by lvignoli** [typst/typst#991] [General]
It is not possible to customize figure caption with the following show rule:

```typ
#show figure.caption: set text(size: 8pt)
```

Compiler output:
```text
error: cannot access fields on type function
   ┌─ common_understanding.typ:57:13
   │
57 │ #show figure.caption: set text()
   │  
```   

What is the best way to achieve the intended result?

Should Typst handle this kind of show rules?

**Comment by LaurenzV** [comment]
Assuming that your figures only consists of pictures (i.e. no code or text), doing `#show figure: set text(8pt)` should be enough to set the size of the caption without affecting anything else. If not, you will have to customize the whole figure using a show rule:

```
#show figure: it => align(center)[
  #it.body
  #text(8pt)[#it.supplement #it.counter.display(): #it.caption]
]
```

**Comment by lvignoli** [comment]
@LaurenzV thanks, I expected that there won't be a much better way than redefining the show rule for the entire environment.

In general, to simplify the rewrite of such "large" show rules with good enough defaults but just some customization, do you think it would be a good idea to have all the defaults show rules accessible somewhere, in the docs for instance?

**Reply by LaurenzV** [reply]
I guess that makes sense. Not sure what the best way to document this would be, though.

**Reply by lvignoli** [reply]
Indeed it seems to weird to just stamp big show rules on each element of the docs.
